ScriptureRenderingPipelineWorker/MergeHandler.cs

```diff
@@ -0,0 +1,205 @@
+using System.Net;
+using Azure.Messaging.ServiceBus;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Extensions.Azure;
+using Microsoft.Extensions.Logging;
+using System.Text.Json;
+using BTTWriterLib;
+using DocumentFormat.OpenXml.Office2010.ExcelAc;
+using Microsoft.Extensions.Configuration;
+using PipelineCommon.Helpers;
+using PipelineCommon.Models.BusMessages;
+using PipelineCommon.Models.ResourceContainer;
+using USFMToolsSharp;
+using USFMToolsSharp.Models.Markers;
+using USFMToolsSharp.Renderers.USFM;
+using YamlDotNet.Serialization;
+
+namespace ScriptureRenderingPipelineWorker;
+
+public class MergeTrigger
+{
+    private readonly ILogger<ProgressReporting> _log;
+    private readonly ServiceBusClient _client;
+    private readonly GiteaClient _giteaClient;
+    private readonly string _destinationUser;
+    private readonly string _giteaBaseAddress;
+    public MergeTrigger(ILogger<ProgressReporting> logger, IAzureClientFactory<ServiceBusClient> serviceBusClientFactory, IConfiguration config)
+    {
+        _log = logger;
+        _client = serviceBusClientFactory.CreateClient("ServiceBusClient");
+        _giteaBaseAddress = config["GiteaBaseAddress"];
+        var user = config["GiteaUser"];
+        var password = config["GiteaPassword"];
+        _destinationUser = config["MergeDestinationUser"];
+        _giteaClient = new GiteaClient(_giteaBaseAddress, user, password);
+    }
+    
+
+    [Function("MergeTrigger")]
+    [ServiceBusOutput("MergedResult", Connection = "ServiceBusConnectionString")]
+    public async Task<MergeResult> MergeRepos([ServiceBusTrigger("MergeRequest", "MergeRequest", IsSessionsEnabled = false, Connection = "ServiceBusConnectionString")] string messageText)
+    {
+        var message = JsonSerializer.Deserialize(messageText, WorkerJsonContext.Default.MergeRequest);
+        if (message == null)
+        {
+            _log.LogError("Got invalid message");
+            return new MergeResult(false, "", null);
```

Comment:
Might be worth passing the error into the MergeResult `message` parameter.

Reply:
The message isn't going to go anywhere that is why I left it blank, but it doesn't hurt for it to be in there.